src/Services/Air/Air.js

```diff
@@ -267,22 +267,46 @@ module.exports = (settings) => {
       'AirRuntimeError.DuplicateTicketFound': async function (ticketNumber) {
         const pnr = await this.getPNRByTicketNumber({ ticketNumber });
         try {
-          return await this.getTicketFromTicketsList(pnr, ticketNumber);
+          const ticket = await this.getTicketFromTicketsList(pnr, ticketNumber);
+
+          if (!ticket) {
+            throw new AirRuntimeError.TicketNotFound({ ticketNumber });
+          }
+
+          return ticket;
         } catch (e) {
           const { splitBookings } = await this.getBooking({ pnr });
-
           if (!splitBookings) {
             throw e;
           }
 
-          const [splitBookingPNR] = splitBookings;
-          const { uapi_ur_locator: urLocator } = await this.getBooking({ pnr: splitBookingPNR });
+          return splitBookings.reduce(async (acc, splitPnr) => {
```

Comment:
I'd like more to use map, return `null` if ticket does not match and then filter results. This way we could get results faster.